### ISSUE TITLE
SimpleList coercion to parent class

### ISSUE THREAD

**Issue by AliciaSchep** [Bioconductor/S4Vectors#4]
Assignment of an object of a class that inherits from a parent class to a class that inherits from SimpleList leads to that object getting coerced to parent class in version 0.17.10 of S4Vectors on R-devel.  Here is an illustrative example:

```r
setClass("DummyClass",
         slots = c(y = "numeric"))

setClass("SpecialClass",
         slots = c(x = "numeric"),
         contains = c("DummyClass"))

setClass("SpecialList",
         prototype = list(elementType = "DummyClass"),
         contains = c("SimpleList"))

x <- new("SpecialList")
new_special <- new("SpecialClass", y = 3, x = 2)
x[["a"]] <- new_special
```
With S4Vectors 0.16.0 the object new_special and x[["a"]] have the same class, "SpecialClass".  With version 0.17.10, x[["a"]] has been coerced into the parent class, "DummyClass" and lost the "x" attribute.  

I am not sure if this coercion to parent class is the intended behavior and this was a deliberate change, or an unintended change, but it is not the behavior I would expect.

**Comment by hpages** [comment]
What a catch! Thanks @AliciaSchep This is indeed a regression I introduced a couple of weeks ago in S4Vectors 0.17.4 when I refactored `[[<-` for List objects (commit 5f54b5fb8a98519af5af137512018c8abb67f7af). Should be fixed in S4Vectors 0.17.12.

**Comment by AliciaSchep** [comment]
Thanks for the quick fix @hpages!
